cub/cub/agent/agent_select_if.cuh

```diff
@@ -825,7 +872,7 @@ struct AgentSelectIf
     {
         // Blocks are launched in increasing order, so just assign one tile per block
         int tile_idx        = (blockIdx.x * gridDim.y) + blockIdx.y; // Current tile index
-        OffsetT tile_offset = tile_idx * TILE_ITEMS; // Global offset for the current tile
+        OffsetT tile_offset = static_cast<OffsetT>(tile_idx) * static_cast<OffsetT>(TILE_ITEMS);
```

Comment:
suggestion: since select now supports large problems, we should expose this at `cub::DeviceSelect` level. I think it's fine to do as a follow up PR, but please, file an issue not to forget this. Same is true for the stencil overload. Please, file an issue to add a stencil API, documentation, and tests.

Reply:
Created [this issue](https://github.com/NVIDIA/cccl/issues/1422) for `OffsetT` in `cub::DeviceSelect`.
Created [this issue](https://github.com/NVIDIA/cccl/issues/1409) to expose new `StencilIf` functionality in `cub::DeviceSelect`.